stackit/internal/services/cdn/distribution/resource.go

```diff
@@ -0,0 +1,612 @@
+
+func mapFields(distribution *cdn.Distribution, model *Model) error {
+	if distribution == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+
+	if distribution.ProjectId == nil {
+		return fmt.Errorf("Project ID not present")
+	}
+
+	if distribution.Id == nil {
+		return fmt.Errorf("CDN distribution ID not present")
+	}
+
+	id := *distribution.ProjectId + core.Separator + *distribution.Id
+
+	model.ID = types.StringValue(id)
+	model.DistributionId = types.StringValue(*distribution.Id)
+	model.ProjectId = types.StringValue(*distribution.ProjectId)
+	model.Status = types.StringValue(*distribution.Status)
+	model.CreatedAt = types.StringValue(distribution.CreatedAt.String())
+	model.UpdatedAt = types.StringValue(distribution.UpdatedAt.String())
+
+	distributionErrors := []attr.Value{}
+	if distribution.Errors != nil {
+		for _, e := range *distribution.Errors {
+			distributionErrors = append(distributionErrors, types.StringValue(*e.En))
+		}
+	}
+	modelErrors, diags := types.ListValue(types.StringType, distributionErrors)
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	model.Errors = modelErrors
+
+	regions := []attr.Value{}
+	for _, r := range *distribution.Config.Regions {
+		regions = append(regions, types.StringValue(string(r)))
+	}
+	modelRegions, diags := types.ListValue(types.StringType, regions)
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	originRequestHeaders := types.MapNull(types.StringType)
+	if distribution.Config.Backend.HttpBackend.OriginRequestHeaders != nil && len(*distribution.Config.Backend.HttpBackend.OriginRequestHeaders) > 0 {
+		headers := map[string]attr.Value{}
+		for k, v := range *distribution.Config.Backend.HttpBackend.OriginRequestHeaders {
+			headers[k] = types.StringValue(v)
+		}
+		mappedHeaders, diags := types.MapValue(types.StringType, headers)
+		originRequestHeaders = mappedHeaders
+		if diags.HasError() {
+			return core.DiagsToError(diags)
+		}
+	}
+	// note that httpbackend is hardcoded here as long as it is the only available backend
+	backend, diags := types.ObjectValue(backendTypes, map[string]attr.Value{
+		"type":                   types.StringValue(*distribution.Config.Backend.HttpBackend.Type),
+		"origin_url":             types.StringValue(*distribution.Config.Backend.HttpBackend.OriginUrl),
+		"origin_request_headers": originRequestHeaders,
+	})
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	cfg, diags := types.ObjectValue(configTypes, map[string]attr.Value{
+		"backend": backend,
+		"regions": modelRegions,
+	})
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	model.Config = cfg
+
+	domains := []attr.Value{}
+	if distribution.Domains != nil {
+		for _, d := range *distribution.Domains {
+			domainErrors := []attr.Value{}
+			if d.Errors != nil {
+				for _, e := range *d.Errors {
+					domainErrors = append(domainErrors, types.StringValue(*e.En))
+				}
```

Comment:
To avoid a nil-pointer exception:
```suggestion
					domainErrors = append(domainErrors, types.StringPointerValue(e.En))
				}
```

Reply:
e.En is a required field in our API, its just the stackit SDK makes everything pointers. I made another change to guard against nil here, please check